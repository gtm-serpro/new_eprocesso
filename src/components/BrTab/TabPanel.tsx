import classNames from "classnames";

export type BrTabPanelProps = {
  children: React.ReactNode;
  index: number;
  activeIndex?: number;
  id?: string;
};

const BrTabPanel = (props: BrTabPanelProps) => {
  const { children, index, activeIndex, ...rest } = props;
  const classes = classNames("tab-panel", { "is-active": index === activeIndex });
  return (
    <div className={classes} {...rest}>
      {children}
    </div>
  );
};

export default BrTabPanel;
