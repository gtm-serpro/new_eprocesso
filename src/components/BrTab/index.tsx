import classNames from "classnames";

export type BrTabProps = {
  children: React.ReactNode;
  dense?: boolean;
  items: string[];
  activeIndex: number;
  onChange: (index: number) => void;
};

const BrTab = (props: BrTabProps) => {
  const { children, dense, items, activeIndex, onChange } = props;

  return (
    <div className={classNames("br-tab", dense && "large")}>
      <nav className="tab-nav">
        <ul>
          {items.length &&
            items.map((item, index) => (
              <li
                className={classNames("tab-item", { "is-active": index === activeIndex })}
                key={index}
              >
                <button type="button" aria-label={item} onClick={() => onChange(index)}>
                  <span className="name">{item}</span>
                </button>
              </li>
            ))}
        </ul>
      </nav>
      <div className="tab-content">{children}</div>
    </div>
  );
};

export default BrTab;
