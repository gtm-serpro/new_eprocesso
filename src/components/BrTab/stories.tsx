import { Meta, Story } from "@storybook/react";
import { useState } from "react";

import BrTab, { BrTabProps } from ".";
import BrTabPanel from "./TabPanel";

const items = ["Panel 1", "Panel 2", "Panel 3"];

export default {
  title: "Navegação/Tab",
  component: BrTab,
  args: { items },
} as Meta;

export const Simples: Story<BrTabProps> = (args) => {
  const [activeTab, setActiveTab] = useState(0);

  return (
    <BrTab {...args} activeIndex={activeTab} onChange={setActiveTab}>
      <BrTabPanel activeIndex={activeTab} index={0}>
        Painel 1
      </BrTabPanel>
      <BrTabPanel activeIndex={activeTab} index={1}>
        Painel 2
      </BrTabPanel>
      <BrTabPanel activeIndex={activeTab} index={2}>
        Painel 3
      </BrTabPanel>
    </BrTab>
  );
};
