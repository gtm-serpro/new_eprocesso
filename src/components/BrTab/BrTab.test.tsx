import { fireEvent, render, screen } from "@testing-library/react";

import BrTab from ".";
import BrTabPanel from "./TabPanel";

describe("<BrTab />", () => {
  const items = ["Panel 1", "Panel 2", "Panel 3"];
  const activeIndex = 0;

  it("deveria renderizar um component Tab com três abas com a primeira habilitada", () => {
    const { container } = render(
      <BrTab items={items} activeIndex={activeIndex} onChange={() => ""}>
        <BrTabPanel activeIndex={activeIndex} index={0}>
          Detalhe Painel 1
        </BrTabPanel>
        <BrTabPanel activeIndex={activeIndex} index={1}>
          Detalhe Painel 2
        </BrTabPanel>
        <BrTabPanel activeIndex={activeIndex} index={2}>
          Detalhe Painel 3
        </BrTabPanel>
      </BrTab>
    );

    expect(screen.getByText(/Painel 1/i)).toBeInTheDocument();
    expect(screen.getByText(/Painel 2/i)).toBeInTheDocument();
    expect(screen.getByText(/Painel 3/i)).toBeInTheDocument();
    expect(screen.getByText(/Detalhe Painel 1/i)).toBeInTheDocument();
    expect(screen.getByText(/Detalhe Painel 2/i)).toBeInTheDocument();
    expect(screen.getByText(/Detalhe Painel 3/i)).toBeInTheDocument();
    expect(container.firstChild).toHaveClass("br-tab");

    expect(container.firstChild).toMatchSnapshot();
  });

  it("deveria renderizar aba como botões e mudar index quando clicar", () => {
    const handleChange = jest.fn();

    render(
      <BrTab items={items} activeIndex={0} onChange={handleChange}>
        <BrTabPanel activeIndex={activeIndex} index={0}>
          Painel 1
        </BrTabPanel>
        <BrTabPanel activeIndex={activeIndex} index={1}>
          Painel 2
        </BrTabPanel>
        <BrTabPanel activeIndex={activeIndex} index={2}>
          Painel 3
        </BrTabPanel>
      </BrTab>
    );

    fireEvent.click(screen.getByRole("button", { name: "Panel 1" }));
    expect(handleChange).toHaveBeenCalledWith(0);

    fireEvent.click(screen.getByRole("button", { name: "Panel 2" }));
    expect(handleChange).toHaveBeenCalledWith(1);

    fireEvent.click(screen.getByRole("button", { name: "Panel 3" }));
    expect(handleChange).toHaveBeenCalledWith(2);
  });
});
