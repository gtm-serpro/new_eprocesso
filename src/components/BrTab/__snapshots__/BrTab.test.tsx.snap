// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrTab /> deveria renderizar um component Tab com trÃªs abas com a primeira habilitada 1`] = `
<div
  class="br-tab"
>
  <nav
    class="tab-nav"
  >
    <ul>
      <li
        class="tab-item is-active"
      >
        <button
          aria-label="Panel 1"
          type="button"
        >
          <span
            class="name"
          >
            Panel 1
          </span>
        </button>
      </li>
      <li
        class="tab-item"
      >
        <button
          aria-label="Panel 2"
          type="button"
        >
          <span
            class="name"
          >
            Panel 2
          </span>
        </button>
      </li>
      <li
        class="tab-item"
      >
        <button
          aria-label="Panel 3"
          type="button"
        >
          <span
            class="name"
          >
            Panel 3
          </span>
        </button>
      </li>
    </ul>
  </nav>
  <div
    class="tab-content"
  >
    <div
      class="tab-panel is-active"
    >
      Detalhe Painel 1
    </div>
    <div
      class="tab-panel"
    >
      Detalhe Painel 2
    </div>
    <div
      class="tab-panel"
    >
      Detalhe Painel 3
    </div>
  </div>
</div>
`;
